File: DailyPractice/Week7/checkIfBST.py

class Solution:

    # Function to check whether a Binary Tree is BST or not.
    val = float("-inf")

    def isBST(self, root):
        # code here
        if root is None:
            return True
        if self.isBST(root.left) is False:
            return False
        if root.data < self.val:
            return False
        else:
            self.val = root.data
        return self.isBST(root.right)

File: DailyPractice/Week7/checkIfBST.cpp
#include<limits.h>
class Node{
    public:
        Node *left;
        Node *right;
        int data;
};
class Solution
{
public:
    // Function to check whether a Binary Tree is BST or not.
    int val = INT_MIN;
    bool isBST(Node *root)
    {
        // Your code here
        if (!root)
            return true;
        if (!isBST(root->left))
            return false;
        if (val > root->data)
            return false;
        else
            val = root->data;
        return isBST(root->right);
    }
};